python/ray/dag/class_node.py

```diff
@@ -225,8 +263,32 @@ def _get_actor_handle(self) -> Optional["ray.actor.ActorHandle"]:
 
     @property
     def num_returns(self) -> int:
-        num_returns = self._bound_options.get("num_returns", None)
-        if num_returns is None:
-            method = self._get_remote_method(self._method_name)
-            num_returns = method.__getstate__()["num_returns"]
-        return num_returns
+        if self.is_class_method_call:
+            num_returns = self._bound_options.get("num_returns", None)
+            if num_returns is None:
+                method = self._get_remote_method(self._method_name)
+                num_returns = method.__getstate__()["num_returns"]
+            return num_returns
+        else:
+            assert self._class_method_output is not None
+            return self._class_method_output.class_method_call.num_returns
+
+    @property
+    def is_class_method_call(self) -> bool:
+        return not self._is_class_method_output
```

Comment:
Can you add a docstring to these methods?

Reply:
Done.